pyiceberg/avro/resolver.py

```diff
@@ -192,7 +195,26 @@ def visit_binary(self, binary_type: BinaryType) -> Writer:
         return BinaryWriter()
 
 
-def resolve(
+CONSTRUCT_WRITER_VISITOR = ConstructWriter()
+
+
+def resolve_writer(
+    struct_schema: Union[Schema, IcebergType],
+    write_schema: Union[Schema, IcebergType],
+) -> Writer:
+    """Resolve the file and read schema to produce a reader.
+
+    Args:
+        struct_schema (Schema | IcebergType): The schema of the Avro file.
+        write_schema (Schema | IcebergType): The requested read schema which is equal, subset or superset of the file schema.
```

Comment:
I think the names and descriptions could be more clear. The description of `struct_schema` is good, but I think it should be `file_schema`.

Reply:
@Fokko, I think the names are still confusing here. When I see `data_schema` I would expect it to be the schema of the data that is being written. And also `write_schema` makes me think of Avro, where a "write schema" is typically the file schema.

Since this is a new concept (writing records into another schema by ignoring some columns), I think it would be good to have really clear names. I'd say `file_schema` and `data_schema`, but if you came to the opposite conclusion about the meaning of `data_schema`, then maybe we should have `record_schema` and `file_schema`? Or possibly `inmemory_schema` and `file_schema`?

Reply:
Also, this says "read schema" in the doc.

Reply:
This is very subjective :D

>  I think the names are still confusing here. When I see data_schema I would expect it to be the schema of the data that is being written.

For me, I would assume that the `data_schema` is in memory. `record_schema` and `file_schema` sounds the most natural to me.